testsuites/listener/shared/client_sg/test_replication.py

```diff
@@ -227,6 +321,92 @@ def test_initial_push_replication(setup_client_syncgateway_test, continuous):
 @pytest.mark.listener
 @pytest.mark.syncgateway
 @pytest.mark.replication
+@pytest.mark.session
+@pytest.mark.parametrize("num_docs, need_attachments, replication_after_backgroundApp", [
+    (100, True, False),
+    (10000, False, False),
+    (10000, False, False)
+])
+def test_push_replication_with_backgroundApp(setup_client_syncgateway_test, num_docs, need_attachments,
+                                             replication_after_backgroundApp):
+    """
+    @summary
+    1. Prepare LiteServ to have specified number of documents.
+    2. Start continous push replication to push the docs into a sync_gateway database.
+    3. While docs are getting replecated , push the app to the background
+    4. Verify if all of the docs get pushed and replication continous when app goes background
+    """
+
+    sg_db = "db"
+    ls_db = "ls_db"
+    seth_channels = ["ABC", "NBC"]
+
+    cluster_config = setup_client_syncgateway_test["cluster_config"]
+    sg_mode = setup_client_syncgateway_test["sg_mode"]
+    ls_url = setup_client_syncgateway_test["ls_url"]
+    sg_one_admin = setup_client_syncgateway_test["sg_admin_url"]
+    sg_one_public = setup_client_syncgateway_test["sg_url"]
+    liteserv = setup_client_syncgateway_test["liteserv"]
+    liteserv_platform = setup_client_syncgateway_test["liteserv_platform"]
+    device_enabled = setup_client_syncgateway_test["device_enabled"]
+
+    sg_config = sync_gateway_config_path_for_mode("listener_tests/listener_tests", sg_mode)
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+
+    # No command to push the app to background on device, so avoid test to run on ios device
+    if((liteserv_platform.lower() != "ios" and liteserv_platform.lower() != "android") or
+       (liteserv_platform.lower() == "ios" and device_enabled)):
+        pytest.skip('This test only valid for mobile')
+    log_info("ls_url: {}".format(ls_url))
+    log_info("sg_one_admin: {}".format(sg_one_admin))
+    log_info("sg_one_public: {}".format(sg_one_public))
+
+    client = MobileRestClient()
+    client.create_user(sg_one_admin, sg_db, "seth", password="password", channels=seth_channels)
+    session = client.create_session(sg_one_admin, sg_db, "seth")
+
+    client.create_database(url=ls_url, name=ls_db)
+    bulk_docs_resp = []
+    if need_attachments:
+        doc_bodies = document.create_docs(
+            doc_id_prefix="seeded_doc",
+            number=num_docs,
+            attachments_generator=attachment.generate_2_png_100_100,
+            channels=seth_channels
+        )
+    else:
+        doc_bodies = document.create_docs(doc_id_prefix='seeded_doc', number=num_docs, channels=seth_channels)
```

Comment:
Remove "seth". Change it to user1 or something like that.

Reply:
done